### PR TITLE
Update System.Reflection.Metadata to v14

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@agocke @tannergooding @jasonmalinowski

**Comment by tannergooding** [comment]
:+1:

File: src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json [hunk 1]
```diff
@@ -1,8 +1,4 @@
 {
-  "supports": {
```
Comment (jasonmalinowski) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: Don't you want this?

**Comment by jasonmalinowski** [comment]
:+1:, other than the removal of the supports sections seemed interesting.

File: src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json (hunk 1 above)
Comment (tmat) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: I don't know what this is supposed to. Currently NuGet reports warnings which tells me that this is useless.

Comment (jasonmalinowski) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: It verifies that the packages you reference actually will run on the stated platforms. Modulo bugs, we should have these in there.

Comment (tmat) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: We probably need to add reference to a package that defines these, in order to enable this validation. I think we'll be able to do that if we move to DNX meta-packages (once we update to IC v37). We can then also avoid manually listing FX dependencies.

Comment (jasonmalinowski) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: The package you list to get the platform definitions isn't something contained in the meta packages IIRC, so you might want to reference them regardless.

Comment (tmat) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: I'm not against adding these if we do so systematically for all projects and it won't cause nuget to spew warnings.

**Pull request closed without merging**